utilities/test_suite/HOST/Tensor_host_audio.cpp

```diff
@@ -0,0 +1,307 @@
+    double cpuTime, wallTime;
+    string testCaseName;
+    printf("\nRunning %s %d times (each time with a batch size of %d images) and computing mean statistics...", func.c_str(), numRuns, batchSize);
+    for (int perfRunCount = 0; perfRunCount < numRuns; perfRunCount++)
+    {
+        for (int iterCount = 0; iterCount < noOfIterations; iterCount++)
+        {
+            for (int cnt = 0; cnt < batchSize; cnt++)
+            {
+                Rpp32f *inputTempF32;
+                inputTempF32 = inputf32 + (cnt * srcDescPtr->strides.nStride);
+
+                SNDFILE	*infile;
+                SF_INFO sfinfo;
+                int	readcount;
+
+                // The SF_INFO struct must be initialized before using it
+                memset (&sfinfo, 0, sizeof (sfinfo));
+                if (!(infile = sf_open (audioFilePath[fileCnt].c_str(), SFM_READ, &sfinfo)))
+                {
+                    sf_close (infile);
+                    continue;
+                }
+
+                srcLengthTensor[cnt] = sfinfo.frames;
+                channelsTensor[cnt] = sfinfo.channels;
+                srcDims[cnt].width = sfinfo.frames;
+                dstDims[cnt].width = sfinfo.frames;
+                srcDims[cnt].height = 1;
+                dstDims[cnt].height = 1;
+
+                int bufferLength = sfinfo.frames * sfinfo.channels;
+                if (inputBitDepth == 2)
+                {
+                    readcount = (int) sf_read_float (infile, inputTempF32, bufferLength);
+                    if (readcount != bufferLength)
+                        cout << "F32 Unable to read audio file completely " << std::endl;
+                }
+                fileCnt++;
+
+                // Close input
+                sf_close (infile);
+            }
+            clock_t startCpuTime, endCpuTime;
+            double startWallTime, endWallTime;
+            switch (testCase)
+            {
+                case 0:
+                {
+                    testCaseName = "non_silent_region_detection";
+                    Rpp32f detectedIndex[batchSize];
+                    Rpp32f detectionLength[batchSize];
+                    Rpp32f cutOffDB = -60.0;
+                    Rpp32s windowLength = 2048;
+                    Rpp32f referencePower = 0.0f;
+                    Rpp32s resetInterval = 8192;
+
+                    startWallTime = omp_get_wtime();
+                    startCpuTime= clock();
+                    if (inputBitDepth == 2)
+                        rppt_non_silent_region_detection_host(inputf32, srcDescPtr, srcLengthTensor, detectedIndex, detectionLength, cutOffDB, windowLength, referencePower, resetInterval, handle);
+                    else
+                        missingFuncFlag = 1;
+
+                    if (testType == 0)
+                        verify_non_silent_region_detection(detectedIndex, detectionLength, testCaseName, batchSize, audioNames, dst);
```

Comment:
The verify_non_silent_region_detection() can't be included inside the time taken to run the function. @sampath1117 this Audio test suite file needs some cleanup, before external PR. Please check.

Reply:
Is it ok if i declare some static buffers for begin and length outside the switch case and have a branch statement in the end of switch case calling two different verify functions based on the case?

Please let me know your thoughts

Reply:
Added a comment stating it won't run in performance tests